### PR TITLE
Extend Span<byte>.IndexOfAny(..., ROS) tests

### PR DESCRIPTION
**Pull request by benaadams**
For https://github.com/dotnet/coreclr/pull/20738

/cc @stephentoub

### CODE REVIEW

File: src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs [hunk 1]
```diff
@@ -80,8 +82,11 @@ public static void DefaultFilledIndexOfTwo_Byte()
                     int index = rnd.Next(0, 2) == 0 ? 0 : 1;
                     byte target0 = targets[index];
                     byte target1 = targets[(index + 1) % 2];
+                    values[0] = target0;
+                    values[1] = target1;
                     int idx = span.IndexOfAny(target0, target1);
                     Assert.Equal(0, idx);
+                    Assert.Equal(idx, span.IndexOfAny(values));
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs]: Nit: it's a bit odd to me that we have two equivalent IndexOfAny calls, but in one case we store into a local and in the other we don't.  I'd prefer to see just:
```C#
Assert.Equal(0, span.IndexOfAny(target0, target1));
Assert.Equal(0, span.IndexOfAny(values));
```
as I think that makes it easier to understand both the correspondence between the two calls, and that for the second one we're explicitly expecting 0 (without my having to understand the previous lines to know that idx == 0).  Plus it's shorter :)

File: src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs [hunk 2]
```diff
@@ -61,12 +61,14 @@ public static void ZeroLengthIndexOfTwo_Byte()
...
             int idx = sp.IndexOfAny<byte>(0, 0);
             Assert.Equal(-1, idx);
+            Assert.Equal(idx, sp.IndexOfAny(new byte[2]));
         }
 
         [Fact]
         public static void DefaultFilledIndexOfTwo_Byte()
         {
             Random rnd = new Random(42);
+            Span<byte> values = new Span<byte>(new byte[2]);
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs]: Do we need to store this into a local, or could we make the test cleaner and easier to understand at the IndexOfAny call site by removing this and changing the below to something like just:
```C#
Assert.Equal(0, span.IndexOfAny(target0, target1));
Assert.Equal(0, span.IndexOfAny(new[] { target0, target1 }.AsSpan()));
```
?

File: src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs [hunk 3]
```diff
@@ -97,29 +102,39 @@ public static void TestMatchTwo_Byte()
                     a[i] = (byte)(i + 1);
                 }
                 ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+                Span<byte> values = new Span<byte>(new byte[2]);
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOfAny.byte.cs]: Same comments/questions for all of these variants.

**Comment by benaadams** [comment]
Simplified tests

**Pull request merged**